common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1464 @@
+ * @param {string|Buffer|URL} newPath destination filepath
+ */
+fs.renameSync = (oldPath, newPath) => {
+	const tiFile = getTiFileFromPathLikeValue(oldPath);
+	// src doesn't actually exist?
+	if (!tiFile.exists()) {
+		const err = noSuchFile('rename', oldPath);
+		err.message = `${err.message} -> '${newPath}'`;
+		err.dest = newPath;
+		throw err;
+	}
+
+	const destFile = getTiFileFromPathLikeValue(newPath);
+	if (destFile.isDirectory()) {
+		// dest is a directory that already exists
+		const err = illegalOperationOnADirectory('rename', oldPath);
+		err.message = `${err.message} -> '${newPath}'`;
+		err.dest = newPath;
+		throw err;
+	}
+
+	let tempPath;
+	if (destFile.isFile()) {
+		// destination file exists, we should overwrite
+		// Our APIs will fail if we try, so first let's make a backup copy and delete the the original
+		tempPath = path.join(fs.mkdtempSync(path.join(Ti.Filesystem.tempDirectory, 'rename-')), path.basename(newPath));
+		destFile.move(tempPath);
+	}
+
+	let success = false;
+	try {
+		success = tiFile.move(newPath);
+	} finally {
+		if (tempPath) {
+			// we temporarily copied the existing destination to back it up...
+			if (success) {
+				// move worked, so we can wipe it away whenever...
+				fs.unlink(tempPath, _err => {});
+			} else {
+				// move it back, because we failed!
+				const tmpFile = getTiFileFromPathLikeValue(tempPath);
+				tmpFile.move(newPath);
+			}
+		}
+	}
+};
+
+/**
+ * @param {string|Buffer|URL} path file path
+ * @param {errorCallback} callback async callback
+ */
+fs.rmdir = (path, callback) => {
+	setTimeout(() => {
+		try {
+			fs.rmdirSync(path);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback();
+	}, 1);
+};
+/**
+ * @param {string|Buffer|URL} path file path
+ */
+fs.rmdirSync = path => {
+	const tiFile = getTiFileFromPathLikeValue(path);
+	if (!tiFile.deleteDirectory(false)) { // do not delete contents!
+		// we failed to delete, but why?
+		// does it exist?
+		if (!tiFile.exists()) {
+			throw noSuchFile('rmdir', path);
+		}
+		// is it a file?
+		if (tiFile.isFile()) {
+			throw notADirectory('rmdir', path);
+		}
+		// is it not empty?
+		const subFiles = tiFile.getDirectoryListing();
+		if (subFiles && subFiles.length > 0) {
+			throw directoryNotEmpty('rmdir', path);
+		}
+	}
+};
+
+/**
+ * @param {string|Buffer|URL} path file path
+ * @param {object} [options] options
+ * @param {boolean} [options.bigint] whether stat values should be bigint
+ * @param {statsCallback} callback async callback
+ */
+fs.stat = (path, options, callback) => {
+	setTimeout(() => {
+		if (typeof options === 'function') {
+			callback = options;
+			options = {};
+		}
+
+		callback(null, new fs.Stats(path));
+	}, 1);
+};
+/**
+ * @param {string|Buffer|URL|integer} path filepath or file descriptor
+ * @param {object} [options] options
+ * @param {boolean} [options.bigint] whether stat values should be bigint
+ * @returns {fs.Stats}
+ */
+fs.statSync = (path, options) => new fs.Stats(path);
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/fs.js;
  Line: 956;
-->

- :warning: common/Resources/ti.internal/extensions/node/fs.js line 956 – 'options' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)